docs/development.md

```diff
@@ -0,0 +1,31 @@
+# Development Guide
+
+This document explains how to develop cluster-api-provide-openstack.
+
+## Using your own openstack-cluster-api-controller image for testing
+
+1. Building your own openstack-cluster-api-controller image
+1. Using your own openstack-cluster-api-controller image
+
+### Building your own openstack-cluster-api-controller image
+
+Set environment variables which is used in Makefile.
+
+Variable | Meaning
+------------ | -------------
+REGISTRY | The registy name 
+VERSION | The image version
+DOCKER_USERNAME | The username for logging in to the Docker registry
+DOCKER_PASSWORD | The password for logging in to the Docker registry
+
+Execute the command to build and upload the image to the Docker registry.
+
+
+```bash
+$ make upload-images
+
```

Comment:
```suggestion
```

Reply:
$make images  can also be used (I am assuming this guide is for setting up dev env)

Reply:
@chaosaffe Thanks, I will fix.
@adisky This section explains building and uploading own images for testing purposes. make images is to build images only. we need to upload images after that. It is better to use upload-images to build and upload.

Reply:
@hidekazuna I understand, but sometime for setting up local env, make images is also used

Reply:
@adisky When do you use the images for setting up local env? I am using make images only for confirming creating images before executing make upload-images.

Reply:
I can't remember clearly, seems currently logic is to drag upstream images
but it's reasonable to use local image to do test , thoughts?

Reply:
@hidekazuna when you do testing for your own code

Reply:
For making motivation clear, I changed section title to `Using your own openstack-cluster-api-controller image for testing cluster creation or deletion`. https://github.com/kubernetes-sigs/cluster-api-provider-openstack/commit/eb9d61a2840a0372328bf65d4778e3d2170b3b22#diff-1bf07101845de6da98f1c948b82c5ce0R5

My understanding is that to test clusterctl create/delete manually, we have to build and upload images to the Docker registry. After that we have to specify the images explicitly in the yaml file. It is not clear for new contributor so that we need some guidance. That is why I proposed this PR.